importlib_resources/_common.py

```diff
@@ -84,25 +83,80 @@ def _(cand: str) -> types.ModuleType:
 
 @resolve.register
 def _(cand: None) -> types.ModuleType:
-    return resolve(_infer_caller().f_globals['__name__'])
+    # PYUPDATE: 3.15 - Update depth & explanation for package_to_anchor()'s removal.
+    # Expected parent stack frames for depth=4:
+    # 0) resolve()'s singledispatch dispatch
+    # 1) resolve()'s singledispatch wrapper
+    # 2) files()
+    # 3) package_to_anchor()
+    # 4) <caller>()
```

Comment:
I don't love this comment. It's nice to know, but it distracts from the function's purpose and represents tech debt. Usually, I like to abstract tech debt into separate modules that provide the compatible and future uses. If we end up sticking with the static depth for detecting the caller, I'd like to see the depth itself encapsulated in a `_caller_depth()` function, whose docstring explains why the number 4 is there, when it might need to change, and _maybe_ a static copy of the current call stack (although I think I'd rather omit that instead ensure there are tests capturing the expectation).
Are there tests that fail if the depth doesn't match the implementation?